x-pack/plugins/ml/server/lib/alerts/register_anomaly_detection_alert_type.ts

```diff
@@ -249,7 +253,9 @@ export function registerAnomalyDetectionAlertType({
       );
 
       const { alertsClient } = services;
-      if (!alertsClient) return { state: {} };
+      if (!alertsClient) {
```

Comment:
When the alerts client is not defined, this is an error that should stop rule exection because alert data cannot be persisted.